chore: new release process ( beta and full ) (#3647)
waku-org/nwaku: .github/ISSUE_TEMPLATE/prepare_beta_release.md

---

<!--
Add appropriate release number to title!

For detailed info on the release process refer to https://github.com/waku-org/nwaku/blob/master/docs/contributors/release-process.md
 -->

### Items to complete

All items below are to be completed by the owner of the given release.

- [ ] Create release branch
- [ ] Assign release candidate tag to the release branch HEAD. e.g. v0.30.0-rc.0
- [ ] Generate and edit releases notes in CHANGELOG.md
- [ ] Review possible update of [config-options](https://github.com/waku-org/docs.waku.org/blob/develop/docs/guides/nwaku/config-options.md)
- [ ] _End user impact_: Summarize impact of changes on Status end users (can be a comment in this issue).

- [ ] **Waku fleets validation**
  - [ ] Lock `waku.test` and `waku.sandbox` fleets to release candidate version
  - [ ] Search _Kibana_ logs from the previous month (since last release was deployed), for possible crashes or errors in `waku.test` and `waku.sandbox`.
      - Most relevant logs are `(fleet: "waku.test" AND message: "SIGSEGV")` OR `(fleet: "waku.sandbox" AND message: "SIGSEGV")`
  - [ ] Unlock `waku.test` and `waku.sandbox` to resume auto-deployment of latest `master` commit

> - [ ] Bump nwaku dependency in [waku-rust-bindings](https://github.com/waku-org/waku-rust-bindings) and make sure all examples and tests work

Author: Addressed!

- [ ] Update according to the new release [waku-compose](https://github.com/waku-org/nwaku-compose) and [waku-simulator](https://github.com/waku-org/waku-simulator).

- [ ] **Proceed with release**

  - [ ] Assign a release tag to the same commit that contains the validated release-candidate tag
  - [ ] Create GitHub release
  - [ ] Deploy the release to DockerHub
  - [ ] Announce the release

- [ ] **Promote release to fleets**.
  - [ ] Update infra config with any deprecated arguments or changed options
  - [ ] [Deploy final release to `waku.sandbox` fleet](https://ci.infra.status.im/job/nim-waku/job/deploy-waku-sandbox)
  - [ ] [Deploy final release to `status.staging` fleet](https://ci.infra.status.im/job/nim-waku/job/deploy-shards-staging/)
  - [ ] [Deploy final release to `status.prod` fleet](https://ci.infra.status.im/job/nim-waku/job/deploy-shards-test/)

- [ ] **Post release**
  - [ ] Submit a PR from the release branch to master. Important to commit the PR with "create a merge commit" option.

>   - [ ] Update waku-org/nwaku-compose with the new release version.

Reviewer: This is mentioned above
```suggestion
```

---
name: Prepare Beta Release
about: Execute tasks for the creation and publishing of a new beta release.
title: 'Prepare beta release 0.0.0'
labels: beta-release
assignees: ''

---

<!--
Add appropriate release number to title!

For detailed info on the release process refer to https://github.com/waku-org/nwaku/blob/master/docs/contributors/release-process.md
 -->

### Items to complete

All items below are to be completed by the owner of the given release.

> - [ ] Create release branch ( e.g. release/v0.X.0-beta ) if it doesn't exist.

Reviewer: ```suggestion
- [ ] Create release branch with major and minor only ( e.g. release/v0.X ) if it doesn't exist.
```

---
name: Prepare release
about: Execute tasks for the creation and publishing of a new release
title: 'Prepare release 0.0.0'
labels: beta-release
assignees: ''

---

<!--
Add appropriate release number to title!

For detailed info on the release process refer to https://github.com/waku-org/nwaku/blob/master/docs/contributors/release-process.md
 -->

### Items to complete

All items below are to be completed by the owner of the given release.

> - [ ] Create release branch ( e.g. release/v0.X.0 )

Author: Actually, it’s fine even without using -beta. If we change the branch name to release/v0.X.0-beta, the release candidate names become unnecessarily long and a bit odd (e.g., v0.X.0-beta-rc.0, v0.X.0-beta-rc.1). But if you still prefer this approach, I’m happy to update it. so beta tag only at final tag after validation complete.

> - [ ] Create release branch ( e.g. release/v0.X.0 )

Reviewer 2: We can actually "upgrade" any beta release to a full release by performing the full round of QA on top of it later. Then it's just about copying the same files that were released with the beta process and removing `beta` from them. There's no need to distinguish a beta and a full release internally as the only difference is the full external QA process.

- [ ] Assign release candidate tag to the release branch HEAD. e.g. v0.X.0-rc.0..N etc.
- [ ] Generate and edit release notes in CHANGELOG.md
- [ ] Review possible updates to [config-options](https://github.com/waku-org/docs.waku.org/blob/develop/docs/guides/nwaku/config-options.md)
- [ ] _End user impact_: Summarize impact of changes on Status end users (can be a comment in this issue).

- [ ] **Waku test and fleets validation**
  - [ ] Ensure all the unit tests (specifically js-waku tests) are green against the release candidate.

>   - [ ] Lock `waku.test` and `waku.sandbox` fleets to the release candidate version. [Need Jenkins access to start the fleet release; if you don't have access, ask the infra team.]

Reviewer: Maybe this point is repeated?

    - Start job on `waku.sandbox` and `waku.test` [ Deployment job ](https://ci.infra.status.im/job/nim-waku/), Wait for completion of the job. If it fails, then debug it.
    - Then lock both fleets to the release candidate version. (If you don't have access to lock fleets, ask the infra team to do it.)
    - Verify at https://fleets.waku.org/ that the fleet is locked to the release candidate version.
    - Check if the image is created at [harbor](https://harbor.status.im/harbor/projects/9/repositories/nwaku/artifacts-tab)

  - [ ] Analyze Kibana logs from the previous month (since the last release was deployed) for possible crashes or errors in `waku.test` and `waku.sandbox`.
      - Most relevant logs are `(fleet: "waku.test" AND message: "SIGSEGV")` OR `(fleet: "waku.sandbox" AND message: "SIGSEGV")`
  - [ ] Unlock `waku.test` and `waku.sandbox` to resume auto-deployment of the latest `master` commit

- [ ] **Proceed with release**

>   - [ ] Assign a release tag ( v0.X.0-beta ) to the same commit that contains the validated release-candidate tag (e.g. v0.X.0-beta) and submit a PR from the release branch to master.

Reviewer: ```suggestion
  - [ ] Assign a release tag ( v0.X.0-beta ) to the same commit that contains the validated release-candidate tag (e.g. v0.X.0-beta-rc.0) and submit a PR from the release branch to master.
```

>   - [ ] Bump nwaku dependency in [waku-rust-bindings](https://github.com/waku-org/waku-rust-bindings) and make sure all examples and tests work

Reviewer: This is needed as well :D
```suggestion
  - [ ] Bump nwaku dependency in [waku-rust-bindings](https://github.com/waku-org/waku-rust-bindings) and make sure all examples and tests work
  - [ ] Bump nwaku dependency in [waku-go-bindings](https://github.com/waku-org/waku-go-bindings) and make sure all tests work
```
